tests/testthat/apps/pws_download_ggplot2/tests/testthat/test-shinytest2.R

```diff
@@ -0,0 +1,105 @@
+library(shinytest2)
+
+# TODO:
+# exportTestValues for this file
+# exportTestValues for pws_click
+# refactor table with settings tests + exportTestValues
+# almost all above for GRID, lattice
+# test warning for setting width too low
+
+test_that("{shinytest2} plot with settings: output types", {
+  skip_on_cran()
+  skip_on_ci()
+  app <- AppDriver$new(name = "pws_download_ggplot2_types", height = 937, width = 1619)
+  vals <- app$get_values()
+
+  # check if outputs are reactive
+  testthat::expect_true(is(vals$export$'plot_r', "reactiveExpr"))
+  testthat::expect_true(is(vals$export$'plot_r', "reactive"))
+  testthat::expect_true(is(vals$export$'plot_r', "function"))
+  for (react_i in vals$export$'plot_data') {
+    testthat::expect_true(is(react_i, "reactive"))
+  }
+
+  # check if plot is ggplot object
+  testthat::expect_true(is(isolate(rlang::inject(vals$export$'plot_r')()), "ggplot"))
```

Comment:
is there a specific reason to use `rlang::inject` here?

Reply:
It is advisable to use class check from `testthat`, for example: `testthat::expect_s3_class(isolate(vals$export$'plot_r'()), "ggplot")`

or use `inherits` prior to the usage of `is`.
![Screenshot from 2022-11-16 14-55-49](https://user-images.githubusercontent.com/86970066/202199419-6f5451fa-eb2c-4a30-92e0-255a7899c8f8.png)

Reply:
actually no, will change to eval